oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLStorageProvider.java

```diff
@@ -66,15 +96,8 @@ public MySQLStorageProvider() {
         return config;
     }
 
-    @Override public void prepare() throws ServiceNotProvidedException, ModuleStartException {
-        Properties settings = new Properties();
-        try {
-            settings.load(ResourceUtils.read("datasource-settings.properties"));
-        } catch (IOException e) {
-            throw new ModuleStartException("load datasource setting file failure.", e);
-        }
-
-        mysqlClient = new JDBCHikariCPClient(settings);
+    @Override public void prepare() throws ServiceNotProvidedException {
+        mysqlClient = new JDBCHikariCPClient(getConfigProperties());
```

Comment:
This is the only benefit of introducing interface `DynamicModuleConfig`, right? I don't think it's worthwhile, WDYT @wu-sheng

Reply:
I have replied, basically, agree with @kezhenxuke. 

@tristaZero  If you want a properties field, read my suggestion.

Reply:
Actually, maybe `DynamicModuleConfig` is not a good name, but i want to say `Dynamic` means we can make the properties of ModuleConfig diverse. For example, like H2StorageConfig, in current H2StorageConfig, only four properties can be set, no more properties. But there are many other properties for `JDBCHikariCPClient`, but H2StorageConfig can not provide.

So look at the `MySQLStorageConfig` implemeting the interface `DynamicModuleConfig`, which only has one member named `properties`, we can put all the of settings into this member, it means `MySQLStorageConfig` can provide any property for `JDBCHikariCPClient`

In future, we don't need to define some important settings as members of `ModuleConfig` previously. Only by implementing `DynamicModuleConfig`, certain `ModuleConfig`, e.g, `ModuleConfig` or `StorageModuleElasticsearchConfig` can provide any setting for their `ModuleProvider`s.

Moreover i think if `ModuleConfig` becomes an interface, that will be better. But this change is big, i don't want to do it togather. And this issue also needs to disscuss in our community.

Reply:
@wu-sheng @kez

Reply:
I know the implementation ia good enough for this feature, but I think @kezhenxkezhand I are arguing about the whole oap design. And these two belong different module, core and mysql storage provider. 
We should not change core just for one module provider, because it is not worth.

Reply:
Hi, you're right. The interface `DynamicModuleConfig` is a big change for he whole oap design. If it's just for `MySQLModuleConfig`, it is not worthy. My initial thought is that for those `ModuleConfig` made up with certain members can implement interface `ModuleConfig`, while other `ModuleConfig` maybe own uncertain members, e.g, the settings of DataSource JDBC can implement interface `DynamicModuleConfig`.

Actually, if `H2ModuleConfig` and `StorageModuleElasticsearchConfig` can implement interface `DynamicModuleConfig`, then the implementation of their `ModuleProvider` will become easier, for you don't worry about how to enumerate and set value for their incalculable properties. But we can consider this issue later.

In future, some of `ModuleConfig` whose properties can not be enumerated easily can also implement `DynamicModuleConfig`.
@wu-sheng @kezhenxu94